File: jplag.frontend.scala/src/main/scala/jplag/scala/Language.scala
package jplag.scala

import java.io.File

import jplag.ProgramI

class Language(val program: ProgramI) extends jplag.Language {
  private val parser = Parser
  this.parser.setProgram(program)

  override def suffixes: Array[String] = {
    Array(".scala")
  }

  override def errorsCount: Int = this.parser.errorsCount

  override def name = "Scala parser"

  override def getShortName = "scala"

  override def min_token_match = 8

  override def parse(dir: File, files: Array[String]): jplag.Structure = this.parser.parse(dir, files)

  override def errors: Boolean = this.parser.getErrors

  override def supportsColumns = true

  override def isPreformated = true

  override def usesIndex = false

  override def noOfTokens: Int = jplag.scala.ScalaTokens.maxId

  override def type2string(typ: Int): String = jplag.scala.ScalaTokens(typ).toString
}

File: jplag.frontend.json/src/main/java/jplag/json/Language.java
package jplag.json;

import java.io.File;

import jplag.ProgramI;

public class Language implements jplag.Language {

    private jplag.json.Parser parser;

    public Language(ProgramI program) {
        this.parser = new jplag.json.Parser();
        this.parser.setProgram(program);
    }

    public String[] suffixes() {
        String[] res = {".json", ".JSON"};
        return res;
    }

    public int errorsCount() {
        return this.parser.errorsCount();
    }

    public String name() {
        return "Json Parser";
    }

    public String getShortName() {
        return "json";
    }

    public int min_token_match() {
        return 2;
    }

    public jplag.Structure parse(File dir, String[] files) {
        return this.parser.parse(dir, files);
    }

    public boolean errors() {
        return this.parser.getErrors();
    }

    public boolean supportsColumns() {
        return true;
    }

    public boolean isPreformated() {
        return true;
    }

    public boolean usesIndex() {
        return false;
    }

    public int noOfTokens() {
        return JsonToken.tokenStructure.table.size();
    }

    public String type2string(int type) {
        return JsonToken.type2string(type);
    }
}
